tests/identities/test_identities_models.py

```diff
@@ -55,6 +57,43 @@ def test_identity_get_all_feature_states_no_segments(
         assert feature_state.enabled is expected
 
 
+def test_update_traits_remove_traits_with_none_value(identity_in_segment):
+    # Given
+    trait = deepcopy(identity_in_segment.identity_traits[0])
+
+    # When
+    trait.trait_value = None
+    identity_in_segment.update_traits([trait])
+    # Then
+    assert identity_in_segment.identity_traits == []
+
+
+def test_update_traits_updates_trait_value(identity_in_segment):
+    # Given
+    trait = deepcopy(identity_in_segment.identity_traits[0])
```

Comment:
As above, I don't really like the `deepcopy` approach - I think it makes the test less readable.